operator/provider_aws.go

```diff
@@ -13,8 +12,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//+build operator_aws
+
 package main
 
 import (
-	_ "github.com/cilium/cilium/pkg/policy/groups/aws" // Register AWS policy group provider.
+	// These dependencies should be included only when this file is included in the build.
+	allocatorAWS "github.com/cilium/cilium/operator/tasks/allocator_aws" // AWS allocator task.
+	_ "github.com/cilium/cilium/pkg/policy/groups/aws"                   // Register AWS policy group provider.
 )
+
+func init() {
+	allocatorProviders["aws"] = &allocatorAWS.AllocatorAWS{}
+}
```

Comment:
This should be replaced with a function that panics if 2 `allocatorProviders` with the same name are being added.

Reply:
This is actually a temporary solution, in the next iteration I'd replace the map with a variable, once we decide what we want to ship in the next release.